src/Umbraco.Core/Strings/DefaultShortStringHelper.cs

```diff
@@ -347,7 +347,8 @@ internal string CleanCodeString(string text, CleanStringType caseType, char sepa
 
             // it's faster to use an array than a StringBuilder
             var ilen = input.Length;
-            var output = new char[ilen * 2]; // twice the length should be OK in all cases
+            var totalSize = ilen * 2;
+            Span<char> output = totalSize <= 1024 ? stackalloc char[totalSize] : new char[totalSize]; // twice the length should be OK in all cases
```

Comment:
Consider renting the char[] from a pool.

Reply:
stackalloc is way faster than a pool. But 1K is too much for a stack, 256 is ideal.

Reply:
> stackalloc is way faster than a pool. But 1K is too much for a stack, 256 is ideal.

Is there a resource about this? While I don't think we will ever allocate 1k, I saw that it was the max available.
I couldn't find anything about the benefits of maxing at 256

Reply:
Check .NET Runtime, they almost all use 256 (unless it requires less): https://github.com/dotnet/runtime/search?p=3&q=stackalloc+256

Reply:
Idea being using the pooled char[] when stackalloc cannot be used.